src/bb/image-io/rt_u3v.h

```diff
@@ -312,224 +286,93 @@ class U3V {
-                            log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20}) [skipped for realtime display]", "", devices_[i].frame_count_);
-                        arv_stream_push_buffer(devices_[i].stream_, bufs[i]);
+                    if (bufs[i] == nullptr) {
+                        log::error("pop_buffer failed  when sync frame due to timeout ({}s)", timeout_us * 1e-6f);
+                        throw ::std::runtime_error("buffer is null");
                     }
+                    devices_[i].frame_count_ = is_gendc_
+                                               ? static_cast<uint32_t>(get_frame_count_from_genDC_descriptor(
+                                    bufs[i], devices_[i]))
+                                               : static_cast<uint32_t>(arv_buffer_get_timestamp(bufs[i]) &
+                                                                       0x00000000FFFFFFFF);
+
+                    i == 0 ?
+                    log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20})", devices_[i].frame_count_,
+                               "") :
+                    log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20})", "",
+                               devices_[i].frame_count_);
                 }
             }
+        }
 
-            //first buffer
-            cameN_idx_ = (cameN_idx_+1) >= num_device ? 0 : cameN_idx_+1;
-            bufs[cameN_idx_] = arv_stream_timeout_pop_buffer (devices_[cameN_idx_].stream_, 30 * 1000 * 1000);
-            if (bufs[cameN_idx_] == nullptr){
-                log::error("pop_buffer(L4) failed due to timeout ({}s)", timeout_us*1e-6f);
-                throw ::std::runtime_error("buffer is null");
-            }
-            devices_[cameN_idx_].frame_count_ = is_gendc_
-                    ? static_cast<uint32_t>(get_frame_count_from_genDC_descriptor(bufs[cameN_idx_], devices_[cameN_idx_]))
-                    : static_cast<uint32_t>(arv_buffer_get_timestamp(bufs[cameN_idx_]) & 0x00000000FFFFFFFF);
-            latest_cnt = devices_[cameN_idx_].frame_count_;
-            cameN_idx_ == 0 ?
-                log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20})", devices_[cameN_idx_].frame_count_, "") :
-                log::trace("All-Popped Frames (USB0, USB1)=({:20}, {:20})", "", devices_[cameN_idx_].frame_count_);
-
-            int internal_count = 0;
-            int max_internal_count = 1000;
+    }
 
-            while (frame_cnt_ >= latest_cnt) {
-                arv_stream_push_buffer(devices_[cameN_idx_].stream_, bufs[cameN_idx_]);
-                bufs[cameN_idx_] = arv_stream_timeout_pop_buffer (devices_[cameN_idx_].stream_, 30 * 1000 * 1000);
-                if (bufs[cameN_idx_] == nullptr){
-                    log::error("pop_buffer(L4) failed due to timeout ({}s)", timeout_us*1e-6f);
+    void setRealTime(std::vector<ArvBuffer *> &bufs, int timeout_us = 3 * 1000 * 1000){
```

Comment:
If the implementation is only for derived class functionality, it should be in the derived class